rfcs/20200619-dropout/README.md

```diff
@@ -0,0 +1,425 @@
+# Proposal for DNNL dropout primitive
+
+## Introduction
+
+The main motivation is request from Amazon mxnet community:
+https://github.com/oneapi-src/oneDNN/issues/656. Currently MXNet is optimized
+with viRngBernoulli from VSL. The problem is that viRngBernoulli returns int*
+where one element has typically 4 bytes. They created new requirement in order
+to reduce memory space of mask. Mask has boolean values {0, 1}, so instead of
+integer representation(4 bytes) it can be stored as 1 bit in bit mask (similar
+optimization is used for specialization of std::vector<bool> in C++ standard
+library). In order to achieve the requirement of storing mask as bit mask, they
+need to perform conversion from table of int to bit mask which reduces memory,
+but has additional performance impact (https://github.com/apache/incubator-mxnet/pull/16735),
+that's why they asked for dropout primitive.
+
+### Current state
+
+Currently in frameworks that use ONE DNN, dropout is implemented as chain of
+elementwise operations.
+
+### Implementation in frameworks
+
+All mainstream frameworks dropout implementations are based on paper:
+http://www.cs.toronto.edu/~rsalakhu/papers/srivastava14a.pdf.
+
+output = input * mask * scale,
+
+where mask is created based on Bernoulli distribution, with probability 1 - rate,
+
+where rate is fraction of input units to drop (real value between 0 and 1),
+usually default value is 0.5,
+
+where scale is 1 / (1 - rate)
+
+#### MXNET
+
+```python
+mxnet.gluon.nn.Dropout(rate, axes=(), **kwargs)
+```
+**Parameters**
```

Comment:
Here is a more detailed description of MXNet Dropout operator: https://mxnet.incubator.apache.org/api/python/docs/api/symbol/symbol.html#mxnet.symbol.Dropout

Reply:
In further commit I will attach note with link,